### ISSUE TITLE
[Regression] Exceptions are not always propagated from `flatMapSequential` operator in Reactor 3.5.5

### ISSUE THREAD

**Issue by wyhasany** [reactor/reactor-core#3474]
The `flatMapSequential` operator in Reactor is used to execute a sequence of suppliers in parallel and collect the results in order. However, in Reactor 3.5.5, it has been observed that exceptions are not always propagated from the `flatMapSequential` operator. This can lead to unexpected behavior, such as failing tests or silent errors.

Probably it was introduced by following PR:
https://github.com/reactor/reactor-core/pull/3397

## Expected Behavior

The `flatMapSequential` operator should always propagate exceptions from the suppliers it executes. This means that if any of the suppliers throw an exception, the `flatMapSequential` operator should propagate that exception to the downstream subscriber.

## Actual Behavior

In Reactor 3.5.5, it has been observed that the flatMapSequential operator does not always propagate exceptions from the suppliers it executes.

## Steps to Reproduce
Here you can find example for reproduction:
https://github.com/wyhasany/reactor-flatmapsequential-regression

```
git clone https://github.com/wyhasany/reactor-flatmapsequential-regression.git
cd reactor-flatmapsequential-regression
./gradlew test
```

This can be seen in the following test:

```java
public static final Scheduler SCHEDULER = Schedulers.fromExecutor(Executors.newFixedThreadPool(4));


@Test
@DisplayName("throws an exception if one flux element fails")
@RepeatedTest(10_000)
void throwsAnExceptionIfOneFluxElementFails() {
    // given
    Supplier<String> successfulSupplier = () -> {
        System.out.println("success1");
        return "success1";
    };
    Supplier<String> successfulSupplier2 = () -> {
        System.out.println("success2");
        return "success2";
    };
    Supplier<String> successfulSupplier3 = () -> {
        System.out.println("success3");
        return "success3";
    };
    Supplier<String> failingSupplier = () -> {
        System.out.println("failing");
        throw new RuntimeException("Failed.");
    };
    var suppliers = Stream.of(successfulSupplier, successfulSupplier2, successfulSupplier3, failingSupplier);

    // when / then
    assertThrows(RuntimeException.class, () -> {
        List<String> strings = runAsync(suppliers, SCHEDULER);
        System.out.println(strings);
    });
}


public static <T> List<T> runAsync(Stream<Supplier<T>> suppliers, Scheduler scheduler) {
    return Flux.fromStream(suppliers)
        .flatMapSequential(supplier ->
            Mono.fromCallable(() -> Optional.ofNullable(supplier.get())).subscribeOn(scheduler)
        )
        .filter(Optional::isPresent)
        .map(Optional::get)
        .collectList()
        .block();
}
```

This test passes in Reactor 3.5.4, but it fails in Reactor 3.5.5. This is because the `flatMapSequential` operator in Reactor 3.5.5 does not propagate the exception from the `failingSupplier`.

## Your Environment

* Reactor version(s) used: 3.5.5
* JVM version (`java -version`): openjdk version "17.0.3" 2022-04-19, OpenJDK Runtime Environment Temurin-17.0.3+7 (build 17.0.3+7), OpenJDK 64-Bit Server VM Temurin-17.0.3+7 (build 17.0.3+7, mixed mode)
* OS and version (eg `uname -a`): Darwin CA040952 21.6.0 Darwin Kernel Version 21.6.0: Mon Aug 22 20:19:52 PDT 2022; root:xnu-8020.140.49~2/RELEASE_ARM64_T6000 arm64

**Comment by kdebski85** [comment]
I confirm the issue with `flatMapSequential`  in `3.5.5` and `3.5.6`. In `3.5.4` it worked fine.

I also tested that `flatMap` works fine in all versions (`3.5.4`, `3.5.5` and `3.5.6`).

**Comment by OlegDokuka** [comment]
fixed with #3477
